docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,455 @@
++ // +kubebuilder:validation:XValidation:rule="!has(self.infrastructureSubnets) || !has(self.defaultGatewayIPs) || !self.defaultGatewayIPs.all(ip, self.infrastructureSubnets.exists(subnet, cidr(subnet).containsIP(ip)))", message="defaultGatewayIPs cannot be a part of infrastructureSubnets"
+type Layer2Config struct {
+// Role describes the network role in the pod.
+//
+// Allowed value is "Secondary".
+// Secondary network is only assigned to pods that use `k8s.v1.cni.cncf.io/networks` annotation to select given network.
+//
+// +kubebuilder:validation:Enum=Primary;Secondary
+// +kubebuilder:validation:Required
+// +required
+Role NetworkRole `json:"role"`
+
+// MTU is the maximum transmission unit for a network.
+// MTU is optional, if not provided, the globally configured value in OVN-Kubernetes (defaults to 1400) is used for the network.
+//
+// +kubebuilder:validation:Minimum=576
+// +kubebuilder:validation:Maximum=65536
+// +optional
+MTU int32 `json:"mtu,omitempty"`
+
+// Subnets are used for the pod network across the cluster.
+// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+//
+// The format should match standard CIDR notation (for example, "10.128.0.0/16").
+// This field must be omitted if `ipam.mode` is `Disabled`.
+//
+// +optional
+Subnets DualStackCIDRs `json:"subnets,omitempty"`
+
++ // excludeSubnets specifies a list of CIDRs removed from the specified CIDRs in `subnets`.
++ // excludeSubnets is optional. When omitted no IP address is excluded and all IP address specified by `subnets` subject to be automatically assigned.
++ // It is still allowed to request IPs from this range through static IP assignment.
++ // Each item should be in range of the specified CIDR(s) in `subnets`.
++ // The maximal exceptions allowed is 25.
++ // The format should match standard CIDR notation (for example, "10.128.0.0/16").
++ // This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
++ // +optional
++ // +kubebuilder:validation:MinItems=1
++ // +kubebuilder:validation:MaxItems=25
++ ExcludeSubnets []CIDR `json:"excludeSubnets,omitempty"`
+
+// JoinSubnets are used inside the OVN network topology.
+//
+// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+// This field is only allowed for "Primary" network.
+// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+// When omitted, the platform will choose a reasonable default which is subject to change over time.
+//
+// +optional
+JoinSubnets DualStackCIDRs `json:"joinSubnets,omitempty"`
+
++ // infrastructureSubnets specifies a list of internal CIDR ranges that OVN-Kubernetes will reserve for internal network infrastructure.
```

Comment:
why is this a subnet ? IMHO, it could be an IP - we can derive the netmask from the subnet provided by the user.

Do we really envision entire ranges of IPs to be reserved for the infrastructure ? Maybe we do. What use cases are there for it ?